kernelci/config/__init__.py

```diff
@@ -24,7 +24,7 @@ def iterate_yaml_files(config_path: str):
     if config_path.endswith('.yaml'):
         yaml_files = [config_path]
     else:
-        yaml_files = glob.glob(os.path.join(config_path, "*.yaml"))
+        yaml_files = glob.glob(os.path.join(config_path, "**", "*.yaml"), recursive=True)
```

Comment:
```suggestion
        yaml_files = glob.iglob(os.path.join(config_path, "**", "*.yaml"), recursive=True)
```

Reply:
While we do not expect a large number of files there, as such memory usage is not a concern, we walk the list of the files we find almost immediately, so we might as well use the variant that returns an iterator.

Reply:
Good catch! Applied

Reply:
Ah, so mypy is not happy now. Sorry, I didn't notice this earlier.

We can do a few things:

- `yaml_files = iter([config_path])` so that the type can be correctly inferred
- `yaml_files: list[str] | Iterator[str] = [config_path]` to add missing type hint

The latter requires an import from `typing`, which adds extra code, sadly.

There is also an option to not use `glob.iglob()` to keep the changes to a minimum.

Reply:
Thanks for suggestions! I went with the former one - aligned `yaml_files` types